pytest/tests/stress/perf_ft_transfer.py

```diff
@@ -0,0 +1,471 @@
+        help='Path to the compiled Fungible Token contract')
+    parser.add_argument('--setup-cluster', default=False,
+        help='Whether to start a dedicated cluster instead of connecting to an existing local node',
+        action='store_true')
+    parser.add_argument('--contracts', default='0,2,4,6,8,a,c,e',
+        help='Number of contract accounts, or alternatively list of subnames, separated by commas')
+    parser.add_argument('--contract-key', default=None,
+        help='Account to deploy contract to and use as source of NEAR for account creation')
+    parser.add_argument('--accounts', default=1000, help='Number of accounts to use')
+    parser.add_argument('--no-account-topup', default=False,
+        action='store_true', help='Fill accounts with additional NEAR prior to testing')
+    parser.add_argument('--shards', default=10, help='number of shards')
+    parser.add_argument('--executors', default=2, help='number of transaction executors')
+    args = parser.parse_args()
+
+    logger.warning(f"SEED is {SEED}")
+    rng = random.Random(SEED)
+
+    if args.setup_cluster:
+        config = cluster.load_config()
+        nodes = cluster.start_cluster(2, 0, args.shards, config, [["epoch_length", 100]], {
+            shard: { "tracked_shards": list(range(args.shards)) }
+            for shard in range(args.shards + 1)
+        })
+        if args.contract_key is None:
+            signer_key = nodes[0].signer_key
+        else:
+            signer_key = key.Key.from_json_file(args.contract_key)
+
+    else:
+        nodes = [
+            cluster.RpcNode("127.0.0.1", 3030),
+        ]
+        # The `nearup` localnet setup stores the keys in this directory.
+        if args.contract_key is None:
+            key_path = (pathlib.Path.home() / ".near/localnet/node0/shard0_key.json").resolve()
+        else:
+            key_path = args.contract_key
+        signer_key = key.Key.from_json_file(key_path)
+
+    ACCOUNTS.append(Account(signer_key))
+    ACCOUNTS[0].refresh_nonce(nodes[0])
+    funding_account = 0
+    start_of_accounts = len(ACCOUNTS) - 1
+    contract_accounts = []
+
+    try:
+        for sub in sorted(rng.sample(string.ascii_lowercase + string.digits, k=int(args.contracts))):
+            funding_key = ACCOUNTS[funding_account].key
+            sub_key = key.Key(f"{sub}.{funding_key.account_id}", funding_key.pk, funding_key.sk)
+            contract_accounts.append(len(ACCOUNTS))
+            ACCOUNTS.append(Account(sub_key))
+    except ValueError:
+        for sub in args.contracts.split(","):
+            funding_key = ACCOUNTS[funding_account].key
+            sub_key = key.Key(f"{sub}.{funding_key.account_id}", funding_key.pk, funding_key.sk)
+            contract_accounts.append(len(ACCOUNTS))
+            ACCOUNTS.append(Account(sub_key))
+
+
+    for i in range(int(args.accounts)):
+        keys = ed25519.create_keypair(entropy=rng.randbytes)
```

Comment:
This only works for python 3.9 and above. Is overriding the default worth that requirement?

Reply:
So the intent here was to have at least some semblance of test reproducibility – if something goes wrong, you can specify the same seed as before and get the same implicit accounts for the test.